module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -3396,21 +3396,43 @@ zfs_link(znode_t *tdzp, znode_t *szp, char *name, cred_t *cr,
 }
 
 static void
-zfs_putpage_commit_cb(void *arg)
+zfs_putpage_sync_commit_cb(void *arg)
 {
 	struct page *pp = arg;
 
 	ClearPageError(pp);
 	end_page_writeback(pp);
 }
 
+static int
+zfs_putpage_async_commit_cb(void *arg)
+{
+	struct page	*pp = arg;
+
+	ClearPageError(pp);
+	end_page_writeback(pp);
+
+	struct inode	*ip = pp->mapping->host;
+	znode_t		*zp = ITOZ(ip);
+	zfsvfs_t	*zfsvfs = ITOZSB(ip);
+
+	ZFS_ENTER(zfsvfs);
+	ZFS_VERIFY_ZP(zp);
+	atomic_dec_32(&zp->z_async_writes_cnt);
+	ZFS_EXIT(zfsvfs);
```

Comment:
We don't need to call any of these `ZFS_*` macro's here since we're guaranteed the inode, and thus znode, will remain valid and because we're not accessing the `zp->z_sa_hdl`.  This function can then return `void`.  This could be simplified to.

```c
static void
zfs_putpage_async_commit_cb(void *arg)
{
        struct page *pp = arg;
        znode_t *zp = ITOZ(pp->mapping->host);

        ClearPageError(pp);
        end_page_writeback(pp);
        atomic_dec_32(&zp->z_async_writes_cnt);
}
```

Reply:
fixed